src/vm/zapsig.cpp

```diff
@@ -786,14 +787,18 @@ TypeHandle ZapSig::DecodeType(Module *pEncodeModuleContext,
                                             level < CLASS_LOADED, // For non-full loads, drop a level when loading generic arguments
                                             NULL,
                                             pZapSigContext);
-                                            
+
+    if (ppRemainingSig != NULL)
+        *ppRemainingSig = p.GetPtr();
+
     return th;
 }
 
 MethodDesc *ZapSig::DecodeMethod(Module *pReferencingModule,
                                  Module *pInfoModule,
                                  PCCOR_SIGNATURE pBuffer,
-                                 TypeHandle * ppTH /*=NULL*/)
+                                 TypeHandle * ppTH, /*=NULL*/
+                                 PCCOR_SIGNATURE* ppRemainingSig /*=NULL*/)
```

Comment:
The `ppRemainingSig` should not be needed. The extra stuff you need for shared generics should be encoded first, not last.